Source/AsyncDisplayKit-Prefix.pch

```diff
@@ -8,23 +8,4 @@
   #import <Foundation/Foundation.h>
 #endif
 
-// Some build systems (Cocoapods, Buck, Bazel, etc) will define these flags manually if the functionality
-// is needed by the app. Carthage in particular, or if a user forgets to set the build flag, benefit from
-// checking if each flag is not defined and then setting it to whether or not the header is accessible.
-
-#ifndef PIN_REMOTE_IMAGE
-// For Carthage or manual builds, this will define PIN_REMOTE_IMAGE if the header is available in the
-// search path e.g. they've dragged in the framework (technically this will not be able to detect if
-// a user does not include the framework in the link binary with build step).
-  #define PIN_REMOTE_IMAGE __has_include(<PINRemoteImage/PINRemoteImage.h>)
-#endif
-
-#ifndef IG_LIST_KIT
-  #define IG_LIST_KIT __has_include(<IGListKit/IGListKit.h>)
-
-  /**
-   * For IGListKit versions < 3.0, you have to use IGListCollectionView.
-   * For 3.0 and later, that class is removed and you use UICollectionView.
-   */
-  #define IG_LIST_COLLECTION_VIEW __has_include(<IGListKit/IGListCollectionView.h>)
-#endif
+#import <AsyncDisplayKit/ASAvailability.h>
```

Comment:
@appleguy The macros need to be publicly visible to the application, at least under Cocoapods. I was very surprised to learn that, but it kind of makes sense – the ASDK header files may either be compiled with ASDK or with the application, depending on who imports them. So if they use those macros, the macros need to be visible in both contexts.

So moving them to ASAvailability makes them publicly visible, and moving ASAvailability to the pch makes them privately visible to all classes.

Can you elaborate on how difficult it is to get the pch configured in your build system? I'm happy to remove ASAvailability from the PCH and put back the explicit imports where needed but my (ignorant?) assumption is that any competent build system could either use the pch as configured in the ASDK build settings or allow you to manually specify a pch file (like Cocoapods).